### PR TITLE
Delegator voting tests

### PR DESCRIPTION
**Pull request by grod220**
Follow up from https://github.com/penumbra-zone/web/pull/1426 & https://github.com/penumbra-zone/web/pull/1427

Now that we have a reasonable mock testing framework, wrote full-coverage tests for delegator voting.

### CODE REVIEW

File: packages/wasm/crate/src/planner.rs [hunk 1]
```diff
@@ -148,8 +148,23 @@ pub async fn plan_transaction(
...
+    let tx_planner_req = TransactionPlannerRequest::decode(request)?;
+    let fvk: FullViewingKey = FullViewingKey::decode(full_viewing_key)?;
+    let fee_asset_id = Id::decode(gas_fee_token)?;
+    let constants: DbConstants = serde_wasm_bindgen::from_value(idb_constants)?;
+    let storage = init_idb_storage(constants).await?;
+
+    let plan = plan_transaction_inner(storage, tx_planner_req, fvk, fee_asset_id).await?;
+
+    Ok(serde_wasm_bindgen::to_value(&plan)?)
+}
```
Comment (grod220) [inline packages/wasm/crate/src/planner.rs:160]: Created `plan_transaction_inner()` in order to separate the serde serialization from the core logic of the function (that gets tested in the test suite).

File: packages/wasm/crate/src/storage.rs [hunk 2]
```diff
@@ -94,8 +94,10 @@ impl<T: Database> Storage<T> {
                 continue;
             }
 
-            if Some(record.note.asset_id()) != asset_id {
-                continue;
+            if let Some(id) = asset_id {
+                if record.note.asset_id() != id {
+                    continue;
+                }
```
Comment (grod220) [inline packages/wasm/crate/src/storage.rs:100]: Bug fix 🐞 : The query should respect passing a `None`. Currently breaks delegator vote planning.

File: packages/wasm/crate/tests/test_get_notes_for_voting.rs [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+    let mock_db = MockDb::new();
+    let tables = get_mock_tables();
+
+    let addr = Address::dummy(&mut OsRng);
+    let value = Value {
+        amount: 10u64.into(),
+        asset_id: asset::Cache::with_known_assets()
+            .get_unit("upenumbra")
+            .unwrap()
+            .id(),
```
Comment (Valentine1898) [inline packages/wasm/crate/tests/test_get_notes_for_voting.rs:80]: It's up to you, but you can simplify it to
``` asset_id: *STAKING_TOKEN_ASSET_ID,```

**Review by Valentine1898 — APPROVED** [review]
LGTM!

File: packages/wasm/crate/tests/test_get_notes_for_voting.rs (hunk 3 above)
Comment (grod220) [reply packages/wasm/crate/tests/test_get_notes_for_voting.rs:80]: oh nice, will update

**Pull request merged**